.github/workflows/integration-tests.yml

```
@@ -0,0 +1,37 @@
+on: [push, pull_request]
```

I'd suggest the following so that we don't double work as much.

```suggestion
on:
  push:
    branches:
      - master
  pull_request
```